src/main/java/nextstep/courses/domain/SessionHeadCount.java

```diff
@@ -0,0 +1,29 @@
+package nextstep.courses.domain;
+
+import java.util.HashSet;
+import java.util.Set;
+import nextstep.users.domain.NsUser;
+
+public class SessionHeadCount {
+
+    private final int maxCount;
+    private final Set<NsUser> nsUsers;
+
+    public SessionHeadCount(int maxCount) {
+        this.maxCount = maxCount;
+        nsUsers = new HashSet<>();
+    }
+
+    public void addPerson(NsUser nsUser) {
+        int curCount = nsUsers.size();
+        nsUsers.add(nsUser);
+
+        if (nsUsers.size() > maxCount) {
+            throw new IllegalArgumentException("수강신청 정원을 넘었습니다.");
+        }
+
+        if (nsUsers.size() == curCount) {
+            throw new IllegalArgumentException("중복 신청입니다.");
+        }
```

Comment:
```suggestion
        if (nsUsers.contains(nsUser)) {
            throw new IllegalArgumentException("중복 신청입니다.");
        }
        
        nsUsers.add(nsUser);
```

위와 같이 모든 예외 처리를 진행한 후 마지막에 콜렉션에 추가하는 방식으로 구현하는 것이 좋지 않을까?